falcon/asgi/ws.py

```diff
@@ -69,8 +71,10 @@ def __init__(
             Union[media.BinaryBaseHandlerWS, media.TextBaseHandlerWS],
         ],
         max_receive_queue: int,
+        default_close_reasons: Dict[Optional[int], str],
     ):
         self._supports_accept_headers = ver != '2.0'
+        self._supports_reason = check_support_reason(ver)
```

Comment:
I think we could use more lazy evaluation when it comes to these checks, and actually check only when needed unless it is a simple check as the `ver != 2.0` above.